aiida/backends/tests/export_and_import.py

```diff
@@ -1481,6 +1483,175 @@ def get_all_node_links(self):
+        export_scenarios_info = None
+
+        def __init__(self, test_case_instance):
+            # Storing a reference to the test case to access the information
+            self.test_case = test_case_instance
+
+            # Information for the size of sub scenarios of every export scenario
+            self.export_scenarios_info = dict()
+            self.export_scenarios_info['default'] = 9
+            self.export_scenarios_info['input_forward_true'] = 3
+            self.export_scenarios_info['create_reversed_false'] = 4
+            self.export_scenarios_info['return_reversed_true'] = 4
+            self.export_scenarios_info['call_reversed_true'] = 2
+
+            # The export flags of every export scenario
+            self.export_scenarios_flags = dict()
+            self.export_scenarios_flags['default'] = dict()
+            self.export_scenarios_flags['input_forward_true'] = {'input_forward': True}
+            self.export_scenarios_flags['create_reversed_false'] = {'create_reversed': False}
+            self.export_scenarios_flags['return_reversed_true'] = {'return_reversed': True}
+            self.export_scenarios_flags['call_reversed_true'] = {'call_reversed': True}
+
+        def construct_complex_graph(self, scenario_name='default', sub_scenario_number=0):
+            """
+            This method creates the graph and returns its nodes.
+            It also returns various combinations of nodes that need to be extracted
+            but also the final expected set of nodes (after adding the expected
+            predecessors, successors etc) based on a set of arguments.
+            :param scenario_name: Various scenarios that we can test. These correspond
+            to the set expansion rules that we would like to override. The available
+            options are: default, input_forward_true, create_reversed_false,
+            return_reversed_true and call_reversed_true.
+            :param sub_scenario_number: Based on the main scenario that we have chosen
+            the sub-scenario (test case) that we would like to get for testing.
+            :return: The graph nodes and the sub-scenario that corresponds to the provided
+            arguments. The sub-scenario is composes of a set of export nodes and a set
+            of nodes that should have been finally exported after applying the set
+            expansion rules.
+            """
+            from aiida.orm.data.base import Int
+            from aiida.orm.calculation.job import JobCalculation
+            from aiida.orm.calculation.work import WorkCalculation
+            from aiida.common.datastructures import calc_states
+            from aiida.common.links import LinkType
+
+            # Node creation
+            d1 = Int(1).store()
+            d2 = Int(2).store()
+            wc1 = WorkCalculation().store()
+            wc2 = WorkCalculation().store()
+
+            pw1 = JobCalculation()
+            pw1.set_computer(self.test_case.computer)
```

Comment:
you use the test_case just to get the computer, correct?
you use the `test_case` just for the computer, correct?
can you just pass the `computer` to the constructor instead?

this is more explicit, while passing the test case creates a loop where the test case creates the ComplexGraph, which again has a reference to the test case, ...

Reply:
Good point, I will do that.
I passed the test_case since I wanted to be more generic.
But it is cleaner what you propose.